src/backend/distributed/sql/udfs/citus_stat_activity/latest.sql

```diff
@@ -0,0 +1,42 @@
+-- citus_stat_activity combines the pg_stat_activity views from all nodes and adds global_pid, nodeid and is_worker_query columns.
+-- The columns of citus_stat_activity don't change based on the Postgres version, however the pg_stat_activity's columns do.
+-- Both Postgres 13 and 14 added one more column to pg_stat_activity (leader_pid and query_id).
+-- citus_stat_activity has the most expansive column set, including the newly added columns.
+-- If citus_stat_activity is queried in a Postgres version where pg_stat_activity doesn't have some columns citus_stat_activity has
+-- the values for those columns will be NULL
+
+CREATE OR REPLACE FUNCTION pg_catalog.citus_stat_activity(OUT global_pid bigint, OUT nodeid int, OUT is_worker_query boolean, OUT datid oid, OUT datname name, OUT pid integer,
+                                                          OUT leader_pid integer, OUT usesysid oid, OUT usename name, OUT application_name text, OUT client_addr inet, OUT client_hostname text,
+                                                          OUT client_port integer, OUT backend_start timestamp with time zone, OUT xact_start timestamp with time zone,
+                                                          OUT query_start timestamp with time zone, OUT state_change timestamp with time zone, OUT wait_event_type text, OUT wait_event text,
+                                                          OUT state text, OUT backend_xid xid, OUT backend_xmin xid, OUT query_id bigint, OUT query text, OUT backend_type text)
+    RETURNS SETOF record
+    LANGUAGE plpgsql
+    AS $function$
+BEGIN
```

Comment:
For future reference: `jsonb_to_recordset` does what we like. If a column (say query_id) isn't available in the jsonb, the return recordset contains `NULL`.